Directory.Build.props

```diff
@@ -10,6 +10,10 @@
   <ItemGroup>
     <PackageReference Include="StyleCop.Analyzers" Version="1.1.0-beta008" PrivateAssets="all" />
   </ItemGroup>
+  
+  <PropertyGroup>
+    <LangVersion>latest</LangVersion>
```

Comment:
>latest [](start = 17, length = 6)

Our build servers actually run a lower version.  We either need to pin a version or we need to ensure EVERYONE including the build servers use the same version.  I think we should pin the version so this change is not right.  It should be 7.0 or 7.1 or whatever

Reply:
@gabog is using some lang constructs in the test framework work that use the latest.  we may need to get that code changed if we want to pin

Reply:
Hi @tomlm, @sgellock,

By setting the LangVersion to latest it will choose the latest version available to the MSBuild that is running on the host. At the present time this means that it will be compiling with C#7.3 for both VS2017 and VS2019 versions of MSBuild. When C#8.0 is released it will compile in 8.0 only if you are building with VS2019, if you have VS2017 it will keep building with C#7.3. 

![image](https://user-images.githubusercontent.com/42191764/60448649-6f20df80-9bfc-11e9-9505-37712931ac26.png)

We believe that the Pipelines build is using VS2019 to run MSBuild, so right now it should be using C#7.3 to compile. We are not sure if you are running another build server with a different configuration.

If you want to pin the version it should be to 7.1 or higher.

Reply:
Some additional background, they changed the default between 2019 and 2017, VS 2017 would default to last major (7.0) VS 2019 defaults to last minor (7.x)
We changed the build agent for the build pipelines to 2019 so it matches what you would have in your machine and building solutions should work on the build server with the latest language version.
There seems to be an issue in the "publish" task that creates the zip for the webdeploy package where it defaults to 7.0. 
Adding the "latest" property explicitly to all the projects fixes that and also allows people to edit the code in VS 2017 and use the latest language features.